pkg/operator/mon/mon.go

```diff
@@ -319,12 +368,72 @@ func (c *Cluster) startPods(mons []*monConfig) error {
 
 	nodeIndex := 0
 	for _, m := range mons {
-		// pick one of the available nodes where the mon will be assigned
-		node := availableNodes[nodeIndex%len(availableNodes)]
+		if _, ok := c.mapping.Node[m.Name]; ok {
+			logger.Debugf("mon %s already assigned to a node, no need to assign", m.Name)
+			continue
+		}
+
+		// when the available nodes are >= than the mon size we just begin taking
+		// "node0" through nodeX (X = c.Size-1)
+		var node v1.Node
+		// when there are enough available nodes for the mon size we just pick the next empty one
+		if len(availableNodes) >= c.Size {
+			for {
+				// pick one of the available nodes where the mon will be assigned
+				node = availableNodes[nodeIndex]
+				if _, ok := c.mapping.Port[node.Name]; ok {
+					nodeIndex++
+				} else {
+					break
+				}
+			}
+		} else {
+			// pick one of the available nodes where the mon will be assigned
+			node = availableNodes[nodeIndex%len(availableNodes)]
+		}
+		logger.Debugf("mon %s assigned to node %s", m.Name, node.Name)
+		nodeInfo, err := getNodeInfoFromNode(node)
+		if err != nil {
+			return fmt.Errorf("couldn't get node info from node %s. %+v", node.Name, err)
+		}
+		if c.HostNetwork {
+			if _, ok := c.mapping.Port[node.Name]; ok {
+				// when the node was already chosen increase port by 1 and set
+				// assignment and that the node was chosen
+				m.Port = c.mapping.Port[node.Name] + int32(1)
```

Comment:
does this only allow two mons on the same node?

Reply:
This allows x amount of mons on the same host, as written in my previous comment I could run all three mons on a minikube (one node) Kubernetes instance.

Reply:
ok i remember you said you could run all of them. If it only adds 1 to the port on this line, how would it increment again for the third mon?

Reply:
The same way for the second mon ;)
When the iteration is running for the third mon, it sees that the entry for the port on the same node as for the second mon, exists and increases the port by one too.

Reply:
oh right, it will be incremented from the last mon. I knew i was missing something simple :)